content/courses/advanced-ada/parts/data_types/arrays.rst

```diff
@@ -525,16 +585,182 @@ we'd like to highlight the following aspects:
+
+    with Measurement_Defs; use Measurement_Defs;
+
+    procedure Show_Measurements is
+       M1, M2  : Measurements (1 .. 10)
+                   := (others => 0.0);
+
+       MD      : Measurements_Derived (1 .. 10);
+       MD2     : Measurements_Derived (1 .. 40);
+       MD10    : Measurements_10;
+    begin
+       M1   := M2;
+       --  ^^^^^^
+       --  Assignment of arrays of
+       --  same type.
+
+       MD   := Measurements_Derived (M1);
+       --      ^^^^^^^^^^^^^^^^^^^^^^^^^
+       --  Conversion to derived type for
+       --  the assignment.
+
+       MD10 := Measurements_10 (M1);
+       --      ^^^^^^^^^^^^^^^^^^^^
+       --  Conversion to derived type for
+       --  the assignment.
+
+       MD10 := Measurements_10 (MD);
+       MD10 := Measurements_10 (MD2 (1 .. 10));
+    end Show_Measurements;
+
+As illustrated by this example, we can assign objects of different array types,
+provided that we perform the appropriate type conversions and make sure that
+the bounds match.
+
+
+.. _Adv_Ada_Array_Subtypes:
+
+Array subtypes
+~~~~~~~~~~~~~~
+
+Naturally, we can also declare subtypes of array types. For example:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Data_Types.Arrays.Derived_Arrays_And_Subtypes.Array_Subtypes
+
+    package Measurement_Defs is
+
+       type Measurements is
+         array (Positive range <>) of Float;
+
+       --
+       --  Simple subtype declaration:
+       --
+       subtype Measurements_Sub is Measurements;
+
+       --
+       --  Subtype with constraints:
+       --
+       subtype Measurements_10 is
+         Measurements (1 .. 10);
+
+       --
+       --  Subtype with dynamic predicate
+       --  (array can only have 20 components
+       --  at most):
+       --
+       subtype Measurements_Max_20 is Measurements
+           with Dynamic_Predicate =>
+                  Measurements_Max_20'Length <= 20;
+
+       --
+       --  Subtype with constraints and
+       --  dynamic predicate (first element
+       --  must be 2.0).
+       --
+       subtype Measurements_First_Two is
+         Measurements (1 .. 10)
+           with Dynamic_Predicate =>
+                  Measurements_First_Two (1) = 2.0;
+
+    end Measurement_Defs;
+
+Here, we're declaring subtypes of the :ada:`Measurements` type. For example,
+:ada:`Measurements_Sub` is a *simple* subtype of :ada:`Measurements` type. In
+the case of the :ada:`Measurements_10` subtype, we're constraining the type to
+a range from 1 to 10.
+
+For the :ada:`Measurements_Max_20` subtype, we're specifying |mdash| via a
+dynamic predicate |mdash| that arrays of this subtype can only have 20
+components at most. Finally, for the :ada:`Measurements_First_Two` subtype,
+we're constraining the type to a range from 1 to 10 and requiring that the
+first component must have a value of 2.0.
+
+Note that we cannot set the default component value for array subtypes |mdash|
+only type declarations are allowed to use that.
```

Comment:
Either "are allowed to do that."
or
"are allowed to use that facility."
or something like that.

Reply:
Indeed, thanks for the correction!